src/Campaigns/Actions/UnarchiveCampaignFormAsPublishStatus.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Give\Campaigns\Actions;
+
+use Give\Campaigns\Models\Campaign;
+use Give\Framework\Database\DB;
+
+/**
+ * @unreleased
+ */
+class UnarchiveCampaignFormAsPublishStatus
+{
+    /**
+     * @unreleased
+     */
+    public function __invoke(Campaign $campaign)
+    {
+        if ($campaign->isDirty('status') &&
+            $campaign->status->isActive() &&
+            $campaign->getOriginal('status')->getValue() === 'archived') {
```

Comment:
I'm _pretty sure_ we can do this:
```suggestion
            $campaign->getOriginal('status')->isArchived()) {
```

Reply:
Yea, that seems to work! 8feeb381ad4016fa2150e3ebdc67333c49b5db2c